### ISSUE TITLE
VarianceNLL Questions

### ISSUE THREAD

**Issue by Miraclerice** [Voldemort108X/hetero_uncertainty#2]
Hi,@Voldemort108X ,I used the acdc dataset for training and found that the loss value of the variance estimator (VarianceNLL) has been rising. I don't know if it is an error in the training script.
In addition, when the registration network and the variance network are input, should the input feature map be separated from the computational graph?as follows：
https://github.com/Voldemort108X/hetero_uncertainty/blob/0ecdac258845ee64ab42e839546c60f8343ccc90/src/voxelmorph/train.py#L306-L307
and https://github.com/Voldemort108X/hetero_uncertainty/blob/0ecdac258845ee64ab42e839546c60f8343ccc90/src/voxelmorph/train.py#L343

Possible modifications are as follows:
```python
input_bank = torch.cat((y_pred[n], y_true[n]), dim=1)
logsigma_image = model_variance(input_bank.detach())
```

**Comment by Miraclerice** [comment]
![Image](https://github.com/user-attachments/assets/ce761ed1-9b42-42fb-ae05-b5186ce4f085)
